contracts/LyraL2/LyraController.sol

```diff
@@ -0,0 +1,211 @@
+                          STRUCTS FOR STORAGE
+  //////////////////////////////////////////////////////////////*/
+
+    /// @notice Params required to close a position
+    /// @dev send these params encoded in bytes
+    /// @param toSettle boolean if true settle position, else close position
+    struct ClosePositionParams {
+        bool toSettle;
+    }
+
+    /// @notice struct indicating the current position
+    /// @param strikeId Strike ID of the option based on strike price
+    /// @param optionType call or put option
+    /// @param amount Amount of sUSD used to purchase option
+    /// @param isActive bool indicating if position is active
+    struct CurrentPosition {
+        uint256 strikeId;
+        uint256 positionId;
+        LyraAdapter.OptionType optionType;
+        uint256 amount;
+        uint256 optionsPurchased;
+        bool isActive;
+    }
+
+    /*///////////////////////////////////////////////////////////////
+                          IMMUTABLES
+  //////////////////////////////////////////////////////////////*/
+
+    IERC20 public constant sUSD =
+        IERC20(0x8c6f28f2F1A3C87F0f938b96d27520d9751ec8d9);
+
+    /*///////////////////////////////////////////////////////////////
+                          MUTABLES
+  //////////////////////////////////////////////////////////////*/
+    /// @notice address of the option market we trade on
+    IOptionMarket public lyraOptionMarket;
+
+    // 0x1f6d98638eee9f689684767c3021230dd68df419
+
+    /// @notice struct inidcating the current position
+    CurrentPosition public currentPosition;
+
+    /// @notice Configures the handlers with base state.
+    /// @param _lyraOptionMarket The option market we buy options on.
+    /// @param _lyraOptionMarket The short collateral contract to settle options.
+    function _configHandler(address _lyraOptionMarket) internal {
+        lyraOptionMarket = IOptionMarket(_lyraOptionMarket);
+    }
+
+    /*///////////////////////////////////////////////////////////////
+                        OPEN / CLOSE LOGIC
+    //////////////////////////////////////////////////////////////*/
+
+    /// @notice Purchases new option on lyra.
+    /// @dev Will use all sUSD balance to purchase option on Lyra.
+    /// @param strikeId Strike ID of the option based on strike price
+    /// @param isCall boolean indication call or put option
+    /// @param amount amount of options to buy
+    /// @param updateExistingPosition boolean indication of if existing position should be updated
+    function _openPosition(
+        uint256 strikeId,
+        bool isCall,
+        uint256 amount,
+        bool updateExistingPosition
+    ) internal returns (LyraAdapter.TradeResult memory tradeResult) {
+        LyraAdapter.OptionType optionType;
+        LyraAdapter.TradeInputParameters memory openPositionParams;
+
+        /// Check if contract doesnt have a position on lyra
+        require(!currentPosition.isActive, "POSITION_ACTIVE");
```

Comment:
prevents the updation of existing position

Reply:
removed this